diagnostic.md

```diff
@@ -6,70 +6,71 @@ Place your responses inside the fenced code-blocks where indicated by comments.
 What are two ways of telling rails where to route a GET request to '/books/4'?
 
 ```rb
-// your response here
+resources :books, only: [:index]
+resources :books, except: [:new, :edit, :show, :create, :update, :destroy]
 ```
 
 In what file does the above code go?
 
 ```md
-// your response here
+config/routes.rb
 ```
 
 In your own words, what does the command `bundle exec rake db:migrate` do?
 
 ```md
-// your response here
+It runs the migrations inside the current bundle
 ```
 
 What is the difference between `rails g scaffold things` and
 `rails g controller things`?
 
 ```md
-// your response here
+`rails g controller things` generates a controller for things, while `rails g scaffold things` generates a model, migration, controller, and views for things.
 ```
 
 What is the job of the `serializer`?
 
 ```md
-// your response here
+The serializer formats the response into proper JSON
 ```
```

Comment:
`The serializer allows you to filter certain data out of your response. The data stil exists, it's just not shown in the response to users.`